### PR TITLE
Yieldmo Adapter: add currency conversion for bid floors

### PR DESCRIPTION
**Pull request by ym-winston**
This branch adds currency conversion for bid floors to the yieldmo adapter. I wasn't able to find any documentation around this feature, so this is largely based off of the implementations found in other adapters.

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java [hunk 1]
```diff
@@ -78,9 +86,31 @@ private ExtImpYieldmo parseImpExt(Imp imp) throws PreBidException {
...
+                ? convertBidFloor(bidFloorPrice, imp.getId(), bidRequest) : bidFloorPrice;
+
+        return imp.toBuilder()
+            .bidfloor(bidFloorPrice.getValue())
+            .bidfloorcur(bidFloorPrice.getCurrency())
+            .ext(mapper.mapper().valueToTree(modifiedExt))
+            .build();
+    }
+
+    private Price convertBidFloor(Price bidFloorPrice, String impId, BidRequest bidRequest) {
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java:103]: This logic id not covered with unit tests

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

**Comment by SerhiiNahornyi** [comment]
Example can be found here
https://docs.prebid.org/prebid-server/developers/add-new-bidder-java.html#currency

File: src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java (hunk 1 above)
Comment (ym-winston) [reply src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java:103]: Tests have now been added, sorry about that.

File: src/test/java/org/prebid/server/bidder/yieldmo/YieldmoBidderTest.java [hunk 2]
```diff
@@ -157,6 +181,53 @@ public void makeHttpRequestShouldReturnCorrectHeaders() {
                         tuple(HttpUtil.ACCEPT_HEADER.toString(), HttpHeaderValues.APPLICATION_JSON.toString()));
     }
 
+    @Test
+    public void makeHttpRequestConvertsCurrencyToUsdWhenAble() {
+        //given
+        final BidRequest bidRequest = givenBidRequest(impBuilder ->
+                impBuilder.bidfloor(BigDecimal.ONE).bidfloorcur("EUR"));
+
+        final var convertedBidFloor = new BigDecimal("1.5");
```
Comment (SerhiiNahornyi) [inline src/test/java/org/prebid/server/bidder/yieldmo/YieldmoBidderTest.java:190]: we are not using vars, only strict types

File: src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java [hunk 3]
```diff
@@ -78,9 +86,31 @@ private ExtImpYieldmo parseImpExt(Imp imp) throws PreBidException {
...
+        final String bidFloorCur = bidFloorPrice.getCurrency();
+        try {
+            final BigDecimal convertedPrice = currencyConversionService
+                    .convertCurrency(bidFloorPrice.getValue(), bidRequest, bidFloorCur, USD_CURRENCY);
+
+            return Price.of(USD_CURRENCY, convertedPrice);
+        } catch (PreBidException e) {
+            //If currency conversion fails, we still want to recieve the bid request.
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java:111]: Nitpick. Typo and space
` // If currency conversion fails, we still want to receive the bid request.`

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java (hunk 3 above)
Comment (ym-winston) [reply src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java]: I have fixed the above issues. Thanks for reviewing.

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**